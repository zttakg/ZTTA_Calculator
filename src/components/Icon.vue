<template>
  <svg v-if="name" :class="['icon', `icon_${name}`]" role="img">
    <title v-if="title">{{ title }}</title>
    <use :xlink:href="`${spritePath}#${name}`"></use>
  </svg>
</template>

<script>
import { SVG_ICONS_SPRITE_PATH } from "@/constants";

export default {
  name: "Icon",

  props: {
    name: String,
    title: String
  },

  data() {
    return {
      spritePath: SVG_ICONS_SPRITE_PATH
    };
  }
};
</script>
