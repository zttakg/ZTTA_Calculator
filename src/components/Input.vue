<template>
  <span :class="classes">
    <span v-if="icon" class="input__icon">
      <Icon :name="icon" />
    </span>
    <input
      class="form-control form-control_type_input input__control"
      :id="name"
      :name="name"
      :type="type"
      :value="value"
      :min="min"
      :max="max"
      :placeholder="placeholder"
      @input="handleInput"
    >
  </span>
</template>

<script>
import { Icon } from "@/components";

export default {
  name: "Input",

  components: {
    Icon
  },

  props: {
    value: {},
    name: String,
    placeholder: String,
    type: {
      type: String,
      default: "text"
    },
    theme: {
      type: String,
      default: "default"
    },
    block: Boolean,
    icon: String,
    min: Number,
    max: Number
  },

  computed: {
    classes() {
      const themes = this.theme
        .split(" ")
        .filter(t => t !== "")
        .map(t => `input_style_${t}`);

      return [
        "input",
        {
          [`input_type_${this.type}`]: this.type,
          [`input_icon`]: this.icon,
          input_block: this.block
        },
        ...themes
      ];
    }
  },

  methods: {
    handleInput(event) {
      this.$emit("input", event.target.value);
    }
  }
};
</script>
