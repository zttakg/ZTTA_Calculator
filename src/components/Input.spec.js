import { shallowMount } from "@vue/test-utils";
import Input from "./Input";
import Icon from "./Icon";

describe("<Input /> component", () => {
  it("should render correct without props", () => {
    const wrapper = shallowMount(Input);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render all props", () => {
    const wrapper = shallowMount(Input, {
      propsData: {
        value: "100",
        name: "test-name",
        type: "number",
        theme: "material rounded",
        placeholder: "test placeholder",
        block: true,
        icon: "test",
        min: 10,
        max: 1000
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render <Icon />", () => {
    const icon = "test";
    const wrapper = shallowMount(Input, {
      propsData: {
        icon
      }
    });
    const iconEl = wrapper.find(Icon);

    expect(iconEl.props().name).toEqual(icon);
  });
});
