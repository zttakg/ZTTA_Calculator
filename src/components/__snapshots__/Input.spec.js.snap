// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> component should render all props 1`] = `
<span class="input input_type_number input_icon input_block input_style_material input_style_rounded"><span class="input__icon"><!----></span>
<input id="test-name" name="test-name" type="number" min="10" max="1000" placeholder="test placeholder" class="form-control form-control_type_input input__control">
</span>
`;

exports[`<Input /> component should render correct without props 1`] = `<span class="input input_type_text input_style_default"><!----> <input type="text" class="form-control form-control_type_input input__control"></span>`;
